telemetry/gparser/parser.py

```diff
@@ -18,13 +20,18 @@ def get_parser(url):
         '^.*dmesg_.+err\.log': DmesgError,
         '^.*dmesg_.+warn\.log': DmesgWarning,
         '^.*enumerated_devs\.log': EnumeratedDevs,
-        '^.*missing_devs\.log': MissingDevs
+        '^.*missing_devs\.log': MissingDevs,
+        '^.*pyadi-iio.*\.xml': [PytestFailure, PytestSkipped, PytestError]
     }
 
     # find parser
     for sk, parser in parsers.items():
         if re.match(sk, url):
-            return parser(url)  
+            if isinstance(parser, list):
+                return [p(url) for p in parser]
+            else:
+                return parser(url)
```

Comment:
The else unnecessary. Should just be:
```python
if isinstance(parser, list):
    return [p(url) for p in parser]
return parser(url)
```
or maybe
```python
parser = parser if isinstance(parser, list) else [parser]
return [p(url) for p in parser]
```

Reply:
https://github.com/sdgtt/telemetry/commit/84f663669ed98fb8e9fe4e5e7eea22438c726ebc